### DISCUSSION TITLE
support for mysql

### DISCUSSION THREAD

**Discussion by alexn-s** [malloydata/malloy#960] [Q&A] [answered]
malloy looks really interesting and i would like to use it with mysql databases.
is there currently support for mysql or planned?

on the homepage only: BigQuery, DuckDB or Postgres (beta) are listed.

**Comment by lloydtabb** [comment, marked as answer]
MySQL support is unlikely, at least soon anyway.  MySQL's SQL is lacks many modern features.  Malloy uses SQL Structs and arrays pretty heavily to achieve nesting.  MySQL lacks these functions.   It also lacks lateral JOINs.

We have a design to support databases without these features through federation, but right now that is just a design.

**Reply by alexn-s** [reply]
thanks for the quick reply and the answer
